File: src/test/java/org/projectodd/nodej/UrlTest.java
package org.projectodd.nodej;

import static org.fest.assertions.Assertions.assertThat;

import org.junit.Ignore;
import org.junit.Test;

public class UrlTest extends NodejTestSupport {

    @Test
    public void UrlParse() {
        eval("var url = require('url')");
        eval("var obj = url.parse('http://user:pass@www.foo.com:8080/some/path?with=querystring&another=value#hash')");
        assertThat(eval("obj.protocol")).isEqualTo("http:");
        assertThat(eval("obj.href")).isEqualTo("http://user:pass@www.foo.com:8080/some/path?with=querystring&another=value#hash");
        assertThat(eval("obj.host")).isEqualTo("www.foo.com:8080");
        assertThat(eval("obj.auth")).isEqualTo("user:pass");
        assertThat(eval("obj.hostname")).isEqualTo("www.foo.com");
        assertThat(eval("obj.port")).isEqualTo("8080");
        assertThat(eval("obj.pathname")).isEqualTo("/some/path");
        assertThat(eval("obj.search")).isEqualTo("?with=querystring&another=value");
        assertThat(eval("obj.path")).isEqualTo("/some/path?with=querystring&another=value");
        assertThat(eval("obj.hash")).isEqualTo("#hash");
    }
}

File: src/test/resources/url/url_test.js
var vertxTest = require('vertx_tests');
var vassert   = vertxTest.vassert;

var url = require('url');


function testParse() {
  var obj = url.parse('http://user:pass@www.foo.com:8080/some/path?with=querystring&another=value#hash')
  vassert.assertEquals(obj.protocol, "http:");
  vassert.assertEquals(obj.href, "http://user:pass@www.foo.com:8080/some/path?with=querystring&another=value#hash");
  vassert.assertEquals(obj.host, "www.foo.com:8080");
  vassert.assertEquals(obj.auth, "user:pass");
  vassert.assertEquals(obj.hostname, "www.foo.com");
  vassert.assertEquals(obj.port, "8080");
  vassert.assertEquals(obj.pathname, "/some/path");
  vassert.assertEquals(obj.search, "?with=querystring&another=value");
  vassert.assertEquals(obj.path, "/some/path?with=querystring&another=value");
  vassert.assertEquals(obj.hash, "#hash");
  vassert.testComplete();
}

vertxTest.startTests(this);
